### PR TITLE
refactor: refactor method getFinancialTransactionByWalletId in Financ…

### PR DESCRIPTION
**Pull request by MackowiakPiotr**

### CODE REVIEW

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionServiceImpl.java [hunk 1]
```diff
@@ -50,9 +50,15 @@ public FinancialTransactionDTO createFinancialTransaction(@Valid CreateFinancial
 
     @Override
     public List<FinancialTransactionDTO> getFinancialTransactionsByWalletId(@Min(1) @NotNull Long walletId) {
-        return financialTransactionRepository.findAllByWalletIdOrderByTransactionDateDesc(walletId).stream()
-                .map(financialTransactionModelMapper::mapFinancialTransactionEntityToFinancialTransactionDTO)
-                .toList();
+        List<FinancialTransaction> financialTransactionsList = financialTransactionRepository.findAllByWalletIdOrderByTransactionDateDesc(walletId);
+
+        if (financialTransactionsList.isEmpty()) {
+            return null;
```
Comment (DawidStuzynski) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionServiceImpl.java:56]: I think that better solution would be to return an empty list

**Review by DawidStuzynski — CHANGES_REQUESTED** [review]
Request changes

**Review by dawciobiel — APPROVED** [review]
Looks good for me

**Review by dawciobiel — APPROVED** [review]
Looks good for me

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionServiceImpl.java [hunk 2]
```diff
@@ -36,33 +37,39 @@ public class FinancialTransactionServiceImpl implements FinancialTransactionServ
...
 
     @Override
     public List<FinancialTransactionDTO> getFinancialTransactionsByWalletId(@Min(1) @NotNull Long walletId) {
-        return financialTransactionRepository.findAllByWalletIdOrderByTransactionDateDesc(walletId).stream()
-                .map(financialTransactionModelMapper::mapFinancialTransactionEntityToFinancialTransactionDTO)
-                .toList();
+        List<FinancialTransaction> financialTransactionsList = financialTransactionRepository
+                .findAllByWalletIdOrderByTransactionDateDesc(walletId);
+
+        if (financialTransactionsList.isEmpty()) {
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionServiceImpl.java:58]: Please remove if-then part of the code at all.

**Review by dawciobiel — CHANGES_REQUESTED** [review]
Request changes

**Review by dawciobiel — CHANGES_REQUESTED** [review]
Request changes

**Review by DawidStuzynski — DISMISSED** [review]
Looks good for me

**Review by dawciobiel — DISMISSED** [review]
Looks good for me

**Review by odzio33 — APPROVED** [review]
Looks good for me

**Review by dawciobiel — APPROVED** [review]
Looks good for me

**Pull request merged**